duckduckgo/Android
# Fire button education flow - experiment #2

Task/Issue URL: https://app.asana.com/0/488551667048375/1197243964666745/f
Tech Design URL: 
CC: 

**Description**:
We are running a second experiment to educate users about the Fire button and its benefits.

This experiment will include some improvements based on the previous experiment.
- Improve copy
- make the animation more prominent
- Animation will run during less amount of time
- New pixels to understand value for users
(full list inside https://app.asana.com/0/488551667048375/1197243964666745/f)

**Steps to test this PR**:

> Scenario 1: testing pixels but not clearing data
1. Hardcode variant `zs`
1. Perform a fresh install
1. Perform a search in DDG
1. Ensure fire button was not highlighted
1. Visit a website with trackers and close CTA
1. Ensure fire button is pulsing
1. Click on the fire button
1. Ensure pixel `m_nav_f_p` with params `fb=true` sent
1. Ensure pixel `m_odc_s` sent and includes `fd` as param
1. Press back / cancel / or click outse to dismiss menu
1. Ensure pixel `m_fdp_c` sent (notice the `p` of promoted)
1. Ensure fire button is not pulsing anymore
1. Press fire button again
1. Ensure pixel `m_nav_f_p` with params `fb=false` sent 
1. Press back / cancel / or click outside to dismiss menu
1. Ensure pixel `m_fd_c`sent (notice there's no `p` of promoted)

> Scenario 2: testing pixels of clearing data
1. Hardcode variant `zs`
1. Perform a fresh install
1. Visit a network website (Instagram)
1. Ensure fire button is highlighted
1. Click on the fire button
1. Ensure pixel `m_nav_f_p` with params `fb=true` sent
1. Ensure pixel `m_odc_s` sent and includes `fd` as param
1. Click on clear all data
1. Ensure pixel `m_fdp_p` sent
1. When app starts again, ensure fire button is not highlighted

> Scenario 3: Pulsing animation only plays for 1h
1. Hardcode variant `zs`
1. Go to `BrowserTabViewModel` line 1712: change the value of `ONE_HOUR_IN_MS` to 6000 (6 seconds)
1. Perform a fresh install
1. Visit a site without trackers (Wikipedia)
1. Ensure fire button is highlighted
1. Wait 6 seconds
1. Ensure animation disappeared
1. click on the fire button
1. Ensure pixel `m_nav_f_p` with params `fb=false` sent
1. Ensure pixel `m_odc_s` sent and includes `fd` as param
1. Press back / cancel / or click outside to dismiss menu
1. Ensure pixel `m_fdp_c` sent (notice the `p` of promoted)

> Scenario 4: Pulsing animation only plays for 1h even if app is killed
1. Hardcode variant `zs`
1. Go to `BrowserTabViewModel` line 1712: change the value of `ONE_HOUR_IN_MS` to 60000 (**this time let's use 60 seconds**)
1. Perform a fresh install
1. Visit a site without trackers (Wikipedia)
1. Ensure the fire button is highlighted
1. The animation will play during 1 minute
1. Kill the app, open the app again...use the browser as usual, but don't click on the fire button
1. Ensure that after 1 minute, fire button is not highlighted anymore

> Scenario 5: Pin toolbar when promoting Fire button
1. Hardcode variant `zs`
1. Perform a fresh install
1. Visit a site without trackers (Wikipedia)
1. Ensure fire button is highlighted
1. Ensure scrolling doesn't hide the toolbar, and pulsing animation always visible
1. Click on the fire button
1. Go back to browser (don't clear data)
1. Ensure toolbar hides and shows following user's scroll

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
